PXB-2729 PXB compilation fails if GTest packages are installed

rahulmalik87 opened the pull request:
https://jira.percona.com/browse/PXB-2729

Fixed the compilation error and xbcloud-t.cc test case

satya-bodapati commented on storage/innobase/xtrabackup/src/xbcloud/xbcloud-t.cc:
```diff
@@ -226,14 +229,14 @@ TEST(s3v4_signer, storageClass) {
+      "SignedHeaders=content-length;content-type;host;x-amz-content-"
+      "sha256;x-amz-date;x-amz-security-token;x-amz-storage-class, "
+      "Signature="
+      "fe6c888f22fb23a7a3fe6f663013b5df3cc761c3777ed4368b325114c885320f");
```
how did you get new signature?

rahulmalik87 replied:
the test was failing and copied the new signature.

satya-bodapati commented on storage/innobase/xtrabackup/src/xbcloud/xbcloud-t.cc:
```diff
@@ -77,7 +78,8 @@ TEST(s3_client, basicDNSv4) {
 
 TEST(s3_client, basicPATHv4) {
```
Should we add similar test for azure APIs?

rahulmalik87 replied:
added

satya-bodapati commented on storage/innobase/xtrabackup/src/xbcloud/CMakeLists.txt:
```diff
@@ -71,7 +71,7 @@ TARGET_LINK_LIBRARIES(xbcloud
   )
 
-IF (WITH_DEBUG)
+IF (WITH_DEBUG AND CMAKE_VERSION VERSION_GREATER 3.22.1)
```
any limitation with cmake < 3.22.1?

rahulmalik87 replied:
yes. GTest::gmock was introduce after 3.22.1 . 

https://cmake.org/cmake/help/v3.23/module/FindGTest.html.

altmannmarcelo reviewed (changes requested):
@rahulmalik87 I think we should not enforce the tests on WITH_DEBUG builds,

We should instead change this flag with WITH_UNIT_TEST and allow it to be turned on/off at will.

Maybe default to OFF on RelWithDebInfo and default to ON on Debug, but with the option to flip it.

rahulmalik87 commented:
> @rahulmalik87 I think we should not enforce the tests on WITH_DEBUG builds,
> 
> We should instead change this flag with WITH_UNIT_TEST and allow it to be turned on/off at will.
> 
> Maybe default to OFF on RelWithDebInfo and default to ON on Debug, but with the option to flip it.

Do you mean adding new CMAKE flag or using existing CMAKE that mysql uses . There is compilation error in using existing CMAKE with_unit_test flag for PXB

altmannmarcelo commented:
> > @rahulmalik87 I think we should not enforce the tests on WITH_DEBUG builds,
> > We should instead change this flag with WITH_UNIT_TEST and allow it to be turned on/off at will.
> > Maybe default to OFF on RelWithDebInfo and default to ON on Debug, but with the option to flip it.
> 
> Do you mean adding new CMAKE flag or using existing CMAKE that mysql uses . There is compilation error in using existing CMAKE with_unit_test flag for PXB

@rahulmalik87 I mean using the existing one. On this issue, we found out that there is no way to compile a Debug version but disable Gunit tests if you have Gunit library installed. We should give the ability to disable its compilation.

rahulmalik87 commented:
> > > @rahulmalik87 I think we should not enforce the tests on WITH_DEBUG builds,
> > > We should instead change this flag with WITH_UNIT_TEST and allow it to be turned on/off at will.
> > > Maybe default to OFF on RelWithDebInfo and default to ON on Debug, but with the option to flip it.
> > 
> > 
> > Do you mean adding new CMAKE flag or using existing CMAKE that mysql uses . There is compilation error in using existing CMAKE with_unit_test flag for PXB
> 
> @rahulmalik87 I mean using the existing one. On this issue, we found out that there is no way to compile a Debug version but disable Gunit tests if you have Gunit library installed. We should give the ability to disable its compilation.

It only compiles with WITH_UNIT_TESTS

satya-bodapati reviewed (approved):
plz squash before merge

altmannmarcelo reviewed (approved):
LGTM

Commit:
PXB-2729 PXB compilation fails if GTest packages are installed
https://jira.percona.com/browse/PXB-2729

Fixed the compilation error and xbcloud-t.cc test case
